presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -2611,6 +2694,19 @@ private void doInsertIntoNewPartition(HiveStorageFormat storageFormat, SchemaTab
             // test rollback
             existingFiles = listAllDataFiles(transaction, tableName.getSchemaName(), tableName.getTableName());
             assertFalse(existingFiles.isEmpty());
+
+            // test statistics
```

Comment:
Is `TestHiveClient` run as a product test?

Can we have more product tests? Something like `INSERT ..` / `CTAS` followed by `SHOW STATS` ? These tests can be optionally both transactional and non-transactional.
(like we have for `SHOW PARTITIONS` --  `TestShowPartitions`)

Reply:
`AbstractTestHiveClient` is a set of `Metastore <-> Hive Connector` integration tests. We run them for all types of supported Metastores, including `FileMetastore`, `HiveMetastore` and FB internal Metastore. So adding a product test won't cover all the cases. From the other side i don't think duplicating test cases is something worth doing. 

Product tests for the statistics were supposed to verify that the real `Hive` can read them, and that the fields used to store them are exactly the same. Not sure if we need to test transactional support as a product test, since different transaction related test cases are already covered as part of an integration test.

Reply:
Testing that hive can correctly read the stats is a good idea.
Testing transactions -- the existing tests are detached from actual usage and they do not convince me that in `BEGIN; INSERT; SELECT` the `SELECT` will correctly see stats during planning. I don't insist however.

Reply:
Transactional support tested in `AbstractTestHiveClient`. It checks whether statistics update is visible within transaction. Whether it is not visible from outside transaction. Whether update is being committed/rollbacked successfully.